Ktisis/Interface/Windows/ActorEdit/EditEquip.cs

```diff
@@ -125,30 +144,152 @@ public class EditEquip {
 				var focus = ImGui.IsWindowFocused() || ImGui.IsWindowHovered();
 
 				ImGui.PushItemWidth(400);
-				ImGui.InputTextWithHint("##equip_search", "Search...", ref ItemSearch, 32);
+				bool pressedEnter = ImGui.InputTextWithHint("##equip_search", "Search...", ref ItemSearch, 32, ImGuiInputTextFlags.EnterReturnsTrue);
+				if (ImGui.IsWindowFocused(ImGuiFocusedFlags.RootAndChildWindows) && !ImGui.IsAnyItemActive() && !ImGui.IsMouseClicked(ImGuiMouseButton.Left))
+					ImGui.SetKeyboardFocusHere(-1);
+
 				ImGui.BeginListBox("##equip_items", new Vector2(-1, 300));
+
 				var items = SlotItems;
 				if (ItemSearch.Length > 0)
-					items = items.Where(i => i.Name.Contains(ItemSearch));
+					items = items.Where(i => i.Name.Contains(ItemSearch, StringComparison.OrdinalIgnoreCase));
+
+				int indexKey = 0;
+				bool isOneSelected = false; // allows one selection per foreach
+				if (LastSelectedItemKey >= items.Count()) LastSelectedItemKey = items.Count() - 1;
+
 				foreach (var item in items) {
 					// TODO: Icon?
-					if (ImGui.Selectable($"{item.Name}")) {
+
+					bool selecting = false;
+
+					selecting |= ImGui.Selectable($"{item.Name}", indexKey == LastSelectedItemKey);
+					if (!isOneSelected)
+					{
+						selecting |= ImGui.IsKeyPressed(KeyBindBrowseUp) && indexKey == LastSelectedItemKey - 1;
+						selecting |= ImGui.IsKeyPressed(KeyBindBrowseDown) && indexKey == LastSelectedItemKey + 1;
+						selecting |= ImGui.IsKeyPressed(KeyBindBrowseUpFast) && indexKey == LastSelectedItemKey - FastScrollLineJump;
+						selecting |= ImGui.IsKeyPressed(KeyBindBrowseDownFast) && indexKey == LastSelectedItemKey + FastScrollLineJump;
+						selecting |= pressedEnter;
+					}
+
+					if (selecting) {
+						if (ImGui.IsKeyPressed(KeyBindBrowseUp) || ImGui.IsKeyPressed(KeyBindBrowseDown) || ImGui.IsKeyPressed(KeyBindBrowseUpFast) || ImGui.IsKeyPressed(KeyBindBrowseDownFast))
+							ImGui.SetScrollY(ImGui.GetCursorPosY() - (ImGui.GetWindowHeight() /2));
 						equip.Id = item.Model.Id;
 						equip.Variant = (byte)item.Model.Variant;
 						Target->Equip(index, equip);
+						LastSelectedItemKey = indexKey;
+						isOneSelected = true;
 					}
 					focus |= ImGui.IsItemFocused();
+					indexKey++;
 				}
 				ImGui.EndListBox();
 				focus |= ImGui.IsItemActive();
 				ImGui.PopItemWidth();
 
-				if (!focus) {
+				if (!focus || ImGui.IsKeyPressed(ImGuiKey.Escape)) {
 					SlotSelect = null;
 					SlotItems = null;
 				}
 			}
 
+			ImGui.End();
+		}
+		public static void DrawControls()
+		{
+			Sets = EquipmentSets.Init(Items);
```

Comment:
And non-null assert here...
```suggestion
			Sets = new EquipmentSets(Items!);
```

Reply:
I ended up changing `ItemsSheet` to non-nullable and asserting when inserting items into this variable 3f9b93b. does that look OK?